### ISSUE TITLE
replace go-astilectron as cross-platform gui

### ISSUE THREAD

**Issue by trembon** [trembon/switch-library-manager#32]
currently all of the astilectron repositories are marked as archived, which will mean they will no longer receive updates and might be removed as well

should implement a replacement for astilectron, maybe wails?

**Comment by firebat20** [comment]
hi trembon,

I agree with moving away from astilectron and moving to wails. Could you test one my builds and give me your opinion on the overall mockup and feel of the new GUI. 

This is just a frontend mockup. I removed the old frontend code and replaced it with Wails and React, the GUI is functionally the same. I have to work on replacing all the astilectron code for all the various calls, to restore function while in GUI mode. Console mode still works perfectly.

See the build attached. 
https://github.com/firebat20/slibar-react/actions/runs/14987509891

Once the calls are updated I can send a PR, maybe to a dev branch?

**Comment by trembon** [comment]
@firebat20 nice work starting on this!
looking at the code it seems like the same goal i had, but i am not able to start the windows version of the build you linked, says something that i need to run it with wails build, so i guess the pipeline build is not complete here?

and one major pain point i wanted to remove was to have an easier debugger set up, but i guess that comes with wails right?

**Comment by firebat20** [comment]
You are right, I missed a couple of steps in the build pipeline. I dont usually use CI/CD, I usually just build/dev locally.

I updated the readme for the screenshots, if you want to peep. 

https://github.com/firebat20/slibar-react

**Comment by trembon** [comment]
no worries, i can look into that later on if that is the case

but otherwise it looks like a good start!
One thing though, i think its better to use a standard framework for the design, just like the current version uses bootstrap, it will be easier for more people to join in and help
maybe shadcn can work? as they seem to have all needed components we can have the need for

i think this can be handled as a version 2 if we continue this work, there are some breaking changes i would like to do as well and this seems like a good time to do it (restructure config file and change logic structure)

**Comment by lubosz** [comment]
Just throwing this astilectron alternative into the mix, after quick research:
https://github.com/wailsapp/wails

**Comment by firebat20** [comment]
The current working model, I tagged a release and updated the screenshots. It is about 85% complete based on function. I didn't want to put together the Organize tab/function just because of possible data loss for myself. I used React and Shadcn with Tailwind. instead of bootstrap. if the preference is still bootstrap, I can replace tailwind in the next go around. See the updated project link. https://github.com/firebat20/based-app . I added a couple of features like light/dark and being able to edit settings file. I also updated a few deprecated functions and other niks. There still a lot of refinement left.

**Comment by trembon** [comment]
@firebat20 looks good, nice work!
i say i would prefer something like shadcn, because then we dont need to spend time to build the components and can focus on the logic
